File: gaia-sdk-javascript/src/graphql/request/input/CreateSkillBuildJobImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class CreateSkillBuildJobImpulse {
public _typeName = "CreateSkillBuildJobImpulse";
    private tenantId:Uuid;
    private skillRef:Uuid;
    private tag:String;

    constructor (tenantId:Uuid, skillRef:Uuid, tag:String) {
        this.tenantId = tenantId;
        this.skillRef = skillRef;
        this.tag = tag;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/CreateSkillBuildJobImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateSkillBuildJobImpulse():
    tenantId: str
    skillRef: str
    tag: str

    def __init__(self, tenantId: str, skillRef: str, tag: str):
        self.tenantId = tenantId
        self.skillRef = skillRef
        self.tag = tag

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.skillRef == other.skillRef and self.tag == other.tag
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'skillRef': self.skillRef, 'tag': self.tag}
